### PR TITLE
🐛(frontend) fix keyboard navigation language picker

### PR DESCRIPTION
**Pull request by daproclaima**
Based on manual acceptance tests and according to issue [#282](https://github.com/numerique-gouv/people/issues/282) comments, more work is needed to finish the keyboard navigation

## Purpose

- Prevent double focus on inner elements of language picker
- Prevent focus on modal elements and instead directly set the visible focus on the elements we decide

## Proposal

- Add a temporary fix to the language picker by manipulating the DOM and giving a tabIndex=-1 to the Cunningham Select component's parent div. This needs to be fixed directly in Cunningham Select as a contribution.
- Do the same to Cunningham Modal

https://github.com/user-attachments/assets/d11faa28-500f-4f0c-8864-1db8ba28b2f1

https://github.com/user-attachments/assets/9dd41c4d-96c1-40ed-b726-3bbb2ecf720c

### CODE REVIEW

File: src/frontend/apps/desk/src/features/language/LanguagePicker.tsx [hunk 1]
```diff
@@ -60,14 +60,29 @@ export const LanguagePicker = () => {
...
+    }
+    const languagePickerDom = document.querySelector(
+      '.c__select-language-picker',
+    );
+
+    if (!languagePickerDom?.firstChild?.firstChild) {
+      return;
+    }
+
+    (languagePickerDom.firstChild.firstChild as HTMLElement).tabIndex = -1;
```
Comment (AntoLC) [inline src/frontend/apps/desk/src/features/language/LanguagePicker.tsx:75]:
```suggestion
    document
      .querySelector('.c__select-language-picker .c__select__wrapper')
      ?.setAttribute('tabindex', '-1');
```

Comment (AntoLC) [reply src/frontend/apps/desk/src/features/language/LanguagePicker.tsx:75]: Can you add a small comment above the useEffect to explain what it is doing ?

File: src/frontend/apps/desk/src/features/mail-domains/components/ModalAddMailDomain.tsx [hunk 2]
```diff
@@ -54,6 +54,30 @@ export const ModalCreateMailDomain = () => {
...
+    const modalContentDom = document.querySelector('.c__modal__content');
+    console.log('modalContentDom: ', modalContentDom); // returns null
+    if (modalContentDom) {
+      (modalContentDom as HTMLElement).tabIndex = -1;
+    }
+  };
+
+  React.useEffect(() => {
+    fixFocusVisible(document);
+  }, []);
```
Comment (AntoLC) [inline src/frontend/apps/desk/src/features/mail-domains/components/ModalAddMailDomain.tsx:79]:
```suggestion
  React.useEffect(() => {
    [
      '.c__modals--opened',
      '.ReactModal__Content',
      '.c__modal__content',
    ].forEach((selector) => {
      document.querySelector(selector)?.setAttribute('tabindex', '-1');
    });
  }, []);
```

Comment (AntoLC) [reply src/frontend/apps/desk/src/features/mail-domains/components/ModalAddMailDomain.tsx:79]: Same maybe a small comment about it?

File: src/frontend/apps/desk/src/features/language/LanguagePicker.tsx (hunk 1 above)
Comment (AntoLC) [reply src/frontend/apps/desk/src/features/language/LanguagePicker.tsx:79]: I think you can simplify, no?

```suggestion
     document  
      .querySelector('.c__select-language-picker .c__select__wrapper')  
      ?.setAttribute('tabindex', '-1'); 
```

File: src/frontend/apps/desk/src/components/__tests__/Modal.test.tsx [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+  );
+};
+
+describe('usePreventFocusVisible hook', () => {
+  test('sets tabindex to -1 on the target elements', () => {
+    render(<TestComponent />);
+
+    const testElement = screen.getByText('Test Element');
+
+    expect(testElement).not.toHaveAttribute('tabindex', '-1');
```
Comment (AntoLC) [inline src/frontend/apps/desk/src/components/__tests__/Modal.test.tsx:25]: This hook is too add a tabindex, no?
It should be that so, no ?: 
```suggestion
    expect(testElement).toHaveAttribute('tabindex', '-1');
```

If this test is complicated with Jest, the next test is maybe enough to assert that your code is working.

Comment (daproclaima) [reply src/frontend/apps/desk/src/components/__tests__/Modal.test.tsx:25]: My bad! I could mock it.

**Pull request merged**